lib/_http_agent.js

```diff
@@ -152,7 +152,12 @@ Agent.prototype.addRequest = function addRequest(req, options, port/*legacy*/,
     options.servername = options.host;
     const hostHeader = req.getHeader('host');
     if (hostHeader) {
-      options.servername = hostHeader.replace(/:.*$/, '');
+      // abc => abc
+      // abc:123 => abc
+      // [::1] => ::1
+      // [::1]:123 => ::1
+      options.servername = hostHeader.replace(/:[^\]]+$/, '')
```

Comment:
I ran a benchmark on your regexp (regexp1), my regexp (regexp2) and a variant that doesn't use regexps (t3). See https://gist.github.com/mattiash/f19f7d862dce54db955267e03319b2d2

The result from my benchmark was that the two regexp-solution was actually faster than your one-regexp-solution, but the fastest was my new variant without any regexp.

```
$ node --version
v8.3.0
$ node index.js
regexp1 x 360,816 ops/sec ±1.15% (88 runs sampled)
regexp2 x 514,166 ops/sec ±1.21% (87 runs sampled)
t3 x 1,195,766 ops/sec ±1.38% (87 runs sampled)
Fastest is t3
```

```
$ node --version
v6.3.1
$ node index.js
regexp1 x 592,158 ops/sec ±0.95% (84 runs sampled)
regexp2 x 758,064 ops/sec ±1.21% (87 runs sampled)
t3 x 829,918 ops/sec ±1.18% (85 runs sampled)
Fastest is t3
``` 

I'll update the PR to use the t3 variant